### PR TITLE
chore: switch over to rules v2

### PR DESCRIPTION
**Pull request by niclim**
This PR:
- switches over src/rulesets/__tests__/end-end.test.ts to the new rule runner code
  - I tried to make the diff here readable, but the result snapshots are generated in different orders so the snapshot is difficult to read; I instead updated it to check for pass / fail conditions so we're getting the overall results, just not the exact shape. It might be better to not have snapshots here since it's a lot of noise to signal (rather you care that everything passed, rather than the actual result shape)
- connects the new rules to the rule runner
- fixes a few bugs found from the end to end tests.

There's currently a failing test in `bulk-compare` end to end tests - it's the [invalid input](https://github.com/snyk/sweater-comb/blob/main/end-end-tests/api-standards/test-bulk.bash#L62-L71) test. This test is unintentionally passing with the old rule runner because (i think) there isn't context passed in here https://github.com/snyk/sweater-comb/blob/main/src/service.ts#L5-L9. The new rule runner doesn't error in this way, since it's only referred to when there's custom context. 

Question for you @cmars - does this test still make sense? It's valid to not have a from, OR to, (and technically valid to not specify either) - I _can_ implement a `if !from && !to` check and throw an error if neither are defined. That being said, I think if we want to get rid of vervet, we'd need to change the interface of this such that we can pass in:
- a glob pattern / list of file paths.
- a way to generate context from the resolved file names

We're planning on doing this shortly after this migration

### CODE REVIEW

File: src/rulesets-new/property-rules.ts [hunk 1]
```diff
@@ -156,12 +156,10 @@ const arrayWithItemsInRequest = new RequestRule({
...
         if (property.raw.type === "array") {
-          for (const item of property.raw.items) {
-            if (!("type" in item)) {
-              throw new RuleError({
-                message: "type was not found array item",
-              });
-            }
+          if (!property.raw.items || !("type" in property.raw.items)) {
+            throw new RuleError({
+              message: "type was not found array item",
```
Comment (cmars) [inline src/rulesets-new/property-rules.ts:161]:
```suggestion
              message: "type was not found array items",
```

**Review by cmars — COMMENTED** [review]
Mostly looks good, one concern over `consistentOperationIds`.

File: src/rulesets-new/operation-rules.ts [hunk 2]
```diff
@@ -105,8 +105,6 @@ const summary = new OperationRule({
 const consistentOperationIds = new OperationRule({
   name: "consistent operation ids",
   docsLink: links.standards.operationSummary,
-  matches: (operation, ruleContext) =>
-    !isBreakingChangeAllowed(ruleContext.custom.changeVersion.stability),
```
[the remark this answers is not in the document]
Comment (niclim) [reply src/rulesets-new/operation-rules.ts:109]: So the old rules enforce this for all endpoints https://github.com/snyk/sweater-comb/blob/dad01f6200e460b1ff8d7df6839fbc503d8715b3/src/rulesets/operations.ts#L97-L104

I can update this and the e2e tests if this is the behavior we want though

[the remark this answers is not in the document]
Comment (cmars) [reply src/rulesets-new/operation-rules.ts:109]: Please do, this was something we missed in our policy change to allow breaking changes in wip & experimental.

[the remark this answers is not in the document]
Comment (niclim) [reply src/rulesets-new/operation-rules.ts]: Done!

File: end-end-tests/api-standards/test.bash [hunk 3]
```diff
@@ -62,6 +62,11 @@ ${COMPARE} \
     --to $HERE/resources/thing/2021-11-10/001-ok-type-change.yaml \
     --context "${CONTEXT}"
 assert_ok
+${COMPARE} \
+    --from $HERE/resources/thing/2021-11-10/000-baseline.yaml \
+    --to $HERE/resources/thing/2021-11-10/001-fail-operationid-change.yaml \
+    --context "${CONTEXT}"
+assert_ok
```
Comment (cmars) [inline end-end-tests/api-standards/test.bash:69]: Passes because stability is experimental. 👍

**Review by cmars — APPROVED** [review]

**Comment by niclim** [comment]
Ok @cmars I upgraded to 0.25.8 which includes a fix to the issue in the main description of the PR (no bulk tests changed).

I'll throw up a PR to clean up the old code + packages, but we can let this code bake for a while before we merge the clean up PR (up to you)

**Pull request merged**